Program.cs

```diff
@@ -48,24 +45,23 @@ public static void AddPerson()
             Console.Write("Please Enter The Person LastName : ");
             var lastName = Console.ReadLine();
 
-            var newPerson = new Person(newName, lastName);
-            var newId = people.Count;
-
-            newPerson.setID(newId + 1);
+            var newId = people.Count + 1;
+            var newPerson = new Person(newId, newName, lastName);
             people.Add(newPerson);
         }
 
         public static void UpdatePerson()
         {
-            PersonList();
-            Console.Write("Please Enter A Person ID : ");
-            var newId = Convert.ToInt32(Console.ReadLine());
-            foreach (var person in people)
+            ListPerson();
+            if (people.Count!=0)
             {
-                if (newId == person.getId())
+                Console.Write("Please Enter The Person ID you want to update : ");
+                string input = Console.ReadLine();
+                bool isInt = Int32.TryParse(input, out var id);
+                if (isInt)
                 {
-                    Console.WriteLine("Person ID : " + person.getId() + ", Name : " + person.getFirstName() +
-                                      ", Lastname : " + person.getLastName());
+                    Console.WriteLine("Person ID : " + people[id].GetId() + ", Name : " + people[id].GetFirstName() +
```

Comment:
You should move it to a method and also should use string interpolation. Because the same code exists in the other places